src/useCases/utils.ts

```diff
@@ -3,14 +3,11 @@ import { pipe } from 'fp-ts/lib/function';
 import { ApiKey } from '../generated/definitions/ApiKey';
 import { Response } from '../domain/types';
 
-// TODO: Get "key-value" from repository or configuration?
-export const onValidApiKey =
-  (apiKey: ApiKey) =>
-  <S extends 200 | 202, T>(valid: Response<S, T>) =>
-    pipe(
-      valid,
-      E.fromPredicate(
-        () => apiKey === 'key-value',
-        () => ({ statusCode: 401 as const, returned: undefined })
-      )
-    );
+export const authorizeApiKey = (apiKey: ApiKey): E.Either<Response<401>, ApiKey> =>
+  pipe(
+    apiKey,
+    E.fromPredicate(
+      () => apiKey === 'key-value',
```

Comment:
At some point we know we have to improve this part considering that the scenario will depend on this value

```suggestion
      // TODO: Get "key-value" from configuration
      () => apiKey === 'key-value',
```